### ISSUE TITLE
Helldivers 2: Unable to join any game/lobby (friends unable to join my lobby as well)

### ISSUE THREAD

**Issue by erkexzcx** [CachyOS/proton-cachyos#1]
# Compatibility Report
- Name of the game with compatibility issues: `Helldivers 2`
- Steam AppID of the game: `553850`

## System Information
- GPU: `AMD Radeon RX 7900 GRE`
- Video driver version: `mesa 1:24.2.6-3`
- Kernel version: `6.11.6-2-cachyos`
- Link to full system information report as [Gist](https://gist.github.com/): `N/A`
- Proton version: `1:9.0.20241031-1`

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms 
Helldivers 2 game works totally fine and I've been playing it for many hours in the past, using `proton-cachyos`. Recently, I did not play it at all and when I tried - I no longer can join any public lobbies, as well as friends lobbies. They cannot join me as well - it feels like there is firewall blocking connection, as it results in timeout. Indefinite waiting to join any public lobby, times out when joining friends lobby etc.

## Reproduction

1. Start `Helldivers 2` with `proton-cachyos`
2. Try to join public lobby (quick play), or join some random game from the map, or join a friend lobby
3. Wait as much as you can
4. Results in error (when joining friend's lobby):

![image](https://github.com/user-attachments/assets/7ed69c3b-4236-4a18-8e76-7b65193390ff)

## Workaround

Switch to `proton-experimental` (installed via Steam) and everything works again.

**Comment by Porak21** [comment]
Same issue for me. Will check if workaround works, thank you.

**Comment by yngvildr-the-voracious** [comment]
I had the same issue. 

Game starts but I'm unable to join neither friends nor random parties on a random planet and Quick Play takes forever.

I reported the issue when on Thursday evening (CET), the situation frustrated me and a discussion with my friends on anti-cheat and Linux reminded me of basic anti-cheat knowledge such as information sent from client mismatching expected server parameters triggers.

That crude, layman assumption made me download Proton Experimental through Steam and relaunche HD2 under it. It worked, my friends and I could play. 

I then made a post on CachyOS forums who recommended gathering logs and posting on a relevant GitHub issue. This looks like one. So I launched the game today and tried to join a public lobby, something I had tried under proton-cachyOS as well without success. The problem was again here, no lobby could be joined. Here are the log files that were gathered during that latest attempt using the guidance I received on the forum (forcing logs at the compatibility layer/proton level)

[steam-553850.log](https://github.com/user-attachments/files/17965633/steam-553850.log)

*I'm doing my part.*

**Comment by Sn0whax** [comment]
Proton-ge-custom-bin was my solution for this

**Comment by ptr1337** [comment]
This is likely that EAC has some kind of libary verification and if steam runtime is used, if the proton-ge-custom is not working, but proton-ge-custom-bin.

**Comment by ventureoo** [comment]
Can you try put this in game properties?
```
STEAMDECK=1 %command%
```

**Comment by theCl0ink** [comment]
> Can you try put this in game properties?
> 
> ```
> STEAMDECK=1 %command%
> ```

This didn't affect the issue of not being able to join other lobbies and it made the game stutter a lot.
As other users i am also able to join lobbies when using other versions of Proton.

**Comment by Snuggerdog** [comment]
Having trouble connecting to public and friend lobbies using Proton Experimental 

Kernel: 6.12.5-2-cachyos
Video Driver: Mesa 24.3.1-cachyos1.6
GPU : AMD ATI Radeon RX 6900 XT 
[
steam Debug Logs ](https://drive.google.com/file/d/1zshW5Go4XHNcRK6TPugYj__hhGU1cXSd/view?usp=drive_link)

IF anyone has any advise or workaround, please kindly share TY !

Also posted in [yngvildr-the-voracious](https://github.com/yngvildr-the-voracious's) forum t[hread](https://discuss.cachyos.org/t/proton-cachyos-possibly-flagged-by-anti-cheat-helldivers-2-on-steam/4459/3)

**Comment by loathingKernel** [comment]
Try with a build from the releases page https://github.com/CachyOS/proton-cachyos/releases. These are built on top of the Steam Linux Runtime.

**Comment by russjr08** [comment]
@loathingKernel I can confirm that the release `proton-cachyos-9.0-20250102-slr` allowed me to join public game lobbies, and has fixed the original issue that was described here. Thanks!

**Comment by loathingKernel** [comment]
Yeah the problem is indeed with anticheat. This is likely unfixable in the regular repo version.

**Comment by Snuggerdog** [comment]
@loathingKernel I was able to get it working using `proton-cachyos-9.0-20250102-slr` using the Linux 6.12.8-zen1-1-zen kernel as it wasn't working using the custom cachy-os one. Thank you and hope this helps someone !

**Comment by loathingKernel** [comment]
So it is an issue with the anti-cheat. There is nothing we can do about it for the in-repo version as it is not using the steam runtime.

I think this can be closed now.
